snapshotter/README.md

```diff
@@ -0,0 +1,141 @@
+# cc-snapshotter
+A containerd snapshotter for confidential containers.
+Currently it adds dmverity protection to container base layers.
+
+## Building
+
+`go` must be installed on the machine and made availabe in the path.
+
+To build, do
+
+```bash
+$ ./scripts/build
+```
+
+It will produce two executables
+
+```bash
+$ ls bin
+cc-snapshotter  ctr-remote
+```
+
+
+`ctr-remote` is an drop-in replacement for `ctr` command that addes the `image rpull` subcommand which pulls a
+given image using cc-snashotter.
+
+## Setup
+
+The snapshotter must be registered with continerd before it can be used.
+
+### Register the snapshotter with containerd
+
+Edit containerd's config file (typically `/etc/containerd/config.toml`) and add the following lines:
+
+```toml
+[proxy_plugins]
+  [proxy_plugins."cc-snapshotter"]
+    type = "snapshot"
+    address = "/var/run/cc-snapshotter.sock"
+
+```
+
+Restart containerd service
+
+``bash
+sudo systemctl restart containerd
+``
+
+Verify that the snapshotter has been registered with containerd.
+```bash
+$ sudo ctr plugins ls
+TYPE                            ID                       PLATFORMS      STATUS
+...
+io.containerd.snapshotter.v1    cc-snapshotter           -              ok
+
+...
+```
+
+### Run the snapshotter service
+
+Make sure that no other instance of the snapshotter is running. Delete the unix socket file if it is left over from a previous run.
+```bash
+$ sudo rm -f /var/run/cc-snapshotter.sock
+
+$ sudo ./bin/cc-snapshotter run
+
+```
+
+
+## Usage
+
+Use another prompt for the following commands.
+
+Use `ctr-remote image rpull` command to pull and image using cc-snapshotter. The image will be pulled consistent with how the image will be pulled in Kubernetes using containerd's CRI plugin.
```

Comment:
Is manual use of ``ctr-remote`` just for testing purposes? Does Kubernetes automatically perform an ``rpull``?